crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -334,86 +225,38 @@ impl AzdlsProfile {
-        test_location: &str,
-    ) -> Result<(), ValidationError> {
-        let table_config = self
-            .generate_table_config(
-                Idents {
-                    warehouse_ident: Uuid::now_v7().into(),
-                    table_ident: table_id.into(),
-                    namespace_ident: ns_id,
-                },
-                &DataAccess {
-                    vended_credentials: false,
-                    remote_signing: false,
-                },
-                test_location,
-                credential,
-                // TODO: This should be a permission based on authz
-                StoragePermissions::ReadWriteDelete,
-            )
-            .await?;
-
-        // TODO: replace with iceberg_rust's FileIO + opendal once sas is available
-        let cred = azure_storage::StorageCredentials::sas_token(
-            table_config
-                .get_custom_prop(self.iceberg_sas_property_key().as_str())
-                .ok_or(CredentialsError::ShortTermCredential {
-                    reason: "Couldn't find sas token in table config.".to_string(),
-                    source: None,
-                })?,
-        )
-        .map_err(|e| CredentialsError::ShortTermCredential {
-            reason: "Error creating azure sas token.".to_string(),
-            source: Some(Box::new(e)),
-        })?;
-        let client =
-            azure_storage_blobs::prelude::BlobServiceClient::new(self.account_name.as_str(), cred);
-        let container = client.container_client(self.filesystem.as_str());
-        let blob = container.blob_client(reduce_scheme_string(test_location, true));
-        let mut get = blob.get().into_stream();
-        while let Some(n) = get.next().await {
-            if let Err(e) = n {
-                return Err(ValidationError::IoOperationFailed(
-                    IoError::FileRead(Box::new(e)),
-                    Box::new(StorageProfile::Azdls(self.clone())),
-                ));
-            }
-        }
-
-        Ok(())
-    }
-
     async fn get_sas_token(
         &self,
-        path: &str,
+        path: &Location,
         cred: StorageCredentials,
         permissions: StoragePermissions,
     ) -> Result<String, CredentialsError> {
         let client =
             azure_storage_blobs::prelude::BlobServiceClient::new(self.account_name.as_str(), cred);
+        let start = time::OffsetDateTime::now_utc();
         let delegation_key = client
             .get_user_deligation_key(
-                // TODO: should we pull in `time` instead?
-                SystemTime::now().into(),
-                SystemTime::now()
-                    .checked_add(std::time::Duration::from_secs(
-                        self.sas_token_validity_seconds.unwrap_or(3600),
+                start,
+                start
+                    .checked_add(time::Duration::seconds(
+                        i64::try_from(self.sas_token_validity_seconds.unwrap_or(3600)).map_err(
```

Comment:
i think we could make this effectively infallible by clamping `self.sas_token_validity_seconds` to i64::MAX

Reply:
Done - we still fail on overflow. This is clearer than using `time::DateTimeOffset::max()` I think?